Создание Бизнес-логики

SAWINBEAST opened the pull request:
Добавлены примерные интерфейсы для сущностей трекера

SAWINBEAST committed:
Интерфейсы Бизнес-логики

Добавлены примерные интерфейсы для сущностей трекера

MAXon28 commented on LifeHabitTracker/LifeHabitTracker.BusinessLogicLayer/INotebook.cs:
```diff
@@ -0,0 +1,39 @@
+    /// <summary>
+    /// Описывает функционал записных сущностей, которые хранят некоторые данные и изредка напоминают о них
+    /// </summary>
+    internal interface INotebook
```
Это пока удали

SAWINBEAST replied:
исправил

MAXon28 commented on LifeHabitTracker/LifeHabitTracker.BusinessLogicLayer/IHabitCreator.cs:
```diff
@@ -0,0 +1,27 @@
+    /// <summary>
+    /// Описывает функционал Создателя и Управляющего Привычек ("HabitFather")
+    /// </summary>
+    internal interface IHabitCreator
```
Тут нет смысла разделять на разные интерфейсы работу с привычкой. Достаточно назвать IHabitService и в этом интерфейсе описать всю работу с привычкой. Пока добавлять много всего не надо, только один метод на добавление привычки.

SAWINBEAST replied:
Исправил

MAXon28 commented on LifeHabitTracker/LifeHabitTracker.BusinessLogicLayer/IHabit.cs:
```diff
@@ -0,0 +1,32 @@
+    /// <summary>
+    /// Описывает функционал Привычки
+    /// </summary>
+    internal interface IHabit
```
Это удали

SAWINBEAST replied:
исправил

MAXon28 commented on LifeHabitTracker/LifeHabitTracker.BusinessLogicLayer/IAnalyst.cs:
```diff
@@ -0,0 +1,27 @@
+    /// <summary>
+    /// Описывает функционал Собирателя всей информации из БД, который делает отчеты
+    /// </summary>
+    internal interface IAnalyst
```
Тоже пока удали

SAWINBEAST replied:
исправил

SAWINBEAST committed:
Тестовый вариант 31.08

Тестовый вариант работы с привычкой

SAWINBEAST committed:
Ничего особенного

Функционал создания привычки еще не создан. Тут просто работа над ошибками

SAWINBEAST committed:
Попытка добавить привычку 5.09

Попытка использования паттерна Хранитель и его использование через Внедрение Зависимостей

SAWINBEAST committed:
Полурабочий вариант 6.09

Почти принимает данные с бота, но лагает.

SAWINBEAST committed:
Рабочий промежуточный вариант

Работает добавление Имени привычки. Начато добавление паттерна Цепочка Обязанностей.

SAWINBEAST committed:
Цепочка обязанностей

Добавлена цепочка обязанностей, но не включены в работу хендлы

SAWINBEAST committed:
В процессе поиска ошибки

Не запускается цепочка

SAWINBEAST committed:
Рабочие запись и вывод привычки

Всё работает без нареканий. Общий интерфейс у классов из Цепочки Зависимостей сохранён.

SAWINBEAST committed:
Аккуратный вид

Убран мусор. Вариант ещё мой. Правки ментора по логике не внесены.

SAWINBEAST committed:
Рабочий костыльный вариант 14.09

Введён паттерн Состояние. Получается ввести привычку и вывести ее из оператива

MAXon28 committed:
LHT-2 Рефакторинг кода

Доработал код процесса создания привычки.
Улучшил стартовую обработку приходящих сообщений боту.

SAWINBEAST committed:
Неоптимальная обработка даты

SAWINBEAST committed:
Рабочий вариант с Датой 29.09

Привычка создается успешно. Дни и время парсятся.

SAWINBEAST committed:
Улучшенная проверка даты 29.09

Добавлены проверки на повтор дня недели + логичности времени (исправлено)

Commit:
LHT-2 Упрощён код

SAWINBEAST committed:
Чистка кода 10.10

MAXon28 commented on LifeHabitTracker/LifeHabitTracker.BusinessLogicLayer/Impls/Chain/DateHandler.cs:
```diff
@@ -0,0 +1,35 @@
+namespace LifeHabitTracker.BusinessLogicLayer.Impls.Chain
+{
+    /// <inheritdoc cref="IDateHandler"/>
+    public class DateHandler : IDateHandler, IDataHandler
```
Всю папку Chain можно удалять

MAXon28 commented on LifeHabitTracker/LifeHabitTracker.BusinessLogicLayer/Impls/State/DateState.cs:
```diff
@@ -0,0 +1,114 @@
+        /// <inheritdoc/>
+        public override (string infoMessage, bool isFinish) HandleData(IContextHabitCreation context, string data, Habit habit)
+        {
+            /// <summary>
```
Внутри метода так делать не надо. И в целом смысла комментировать это нет.

MAXon28 commented on LifeHabitTracker/LifeHabitTracker.BusinessLogicLayer/Impls/State/DateState.cs:
```diff
@@ -0,0 +1,114 @@
+
+            if (daysAndTime[0] != null && daysAndTime[1] != null)
+            {
+                /// <summary>
```
Внутри метода так делать не надо. И в целом смысла комментировать это нет.

MAXon28 commented on LifeHabitTracker/LifeHabitTracker.BusinessLogicLayer/Impls/State/DateState.cs:
```diff
@@ -0,0 +1,114 @@
+                            $"\nВы также можете выбрать \"Будни\",\"Выходные\",\"Ежедневно\"." +
+                            $"\nВведите дни и время заново, пожалуйста :)", false);
+
+                /// <summary>
```
Внутри метода так делать не надо. И в целом смысла комментировать это нет.

MAXon28 commented on LifeHabitTracker/LifeHabitTracker.BusinessLogicLayer/Impls/State/DateState.cs:
```diff
@@ -0,0 +1,114 @@
+        /// </summary>
+        /// <param name="resultTime">Перечисление времени через запятую</param>
+        /// <param name="times">Отредактированный массив времени</param>
+        /// <returns></returns>
```
Что возвращает?

MAXon28 commented on LifeHabitTracker/LifeHabitTracker.BusinessLogicLayer/Impls/State/DateState.cs:
```diff
@@ -0,0 +1,114 @@
+        /// </summary>
+        /// <param name="resultDay">Перечисление дней через запятую</param>
+        /// <param name="days">Отредактированный массив дней недели</param>
+        /// <returns></returns>
```
Что возвращает?

MAXon28 commented on LifeHabitTracker/LifeHabitTracker.BusinessLogicLayer/Impls/State/DescState.cs:
```diff
@@ -0,0 +1,29 @@
+
+namespace LifeHabitTracker.BusinessLogicLayer.Impls.State
+{
+    public class DescState : HabitCreationState
```
Комментарий к классу.

MAXon28 commented on LifeHabitTracker/LifeHabitTracker.BusinessLogicLayer/Impls/State/TypeState.cs:
```diff
@@ -0,0 +1,44 @@
+        /// <inheritdoc/>
+        public override (string infoMessage, bool isFinish) HandleData(IContextHabitCreation context, string data, Habit habit)
+        {
+            /// <summary>
```
Что возвращает?

MAXon28 commented on LifeHabitTracker/LifeHabitTracker.BusinessLogicLayer/Interfaces/Chain/IReciever.cs:
```diff
@@ -0,0 +1,47 @@
+    /// <summary>
+    /// Функционал хранителя состояний данных привычки
+    /// </summary>
+    public interface IReciever
```
Удаляем интерфейс.

MAXon28 commented on LifeHabitTracker/LifeHabitTracker.BusinessLogicLayer/Interfaces/Chain/INameHandler.cs:
```diff
@@ -0,0 +1,26 @@
+    /// <summary>
+    /// Функционал обработчика событий
+    /// </summary>
+    public interface INameHandler
```
Удаляем интерфейс.

MAXon28 commented on LifeHabitTracker/LifeHabitTracker.BusinessLogicLayer/Interfaces/Chain/IDescHandler.cs:
```diff
@@ -0,0 +1,25 @@
+    /// <summary>
+    /// Функционал обработчика событий
+    /// </summary>
+    public interface IDescHandler
```
Удаляем интерфейс.

MAXon28 commented on LifeHabitTracker/LifeHabitTracker.BusinessLogicLayer/Interfaces/Chain/IDateHandler.cs:
```diff
@@ -0,0 +1,25 @@
+    /// <summary>
+    /// Функционал обработчика событий
+    /// </summary>
+    public interface IDateHandler
```
Удаляем интерфейс.

MAXon28 commented on LifeHabitTracker/LifeHabitTracker.BusinessLogicLayer/Interfaces/Chain/IDataHandler.cs:
```diff
@@ -0,0 +1,22 @@
+    /// <summary>
+    /// Общий Функционал обработчика событий
+    /// </summary>
+    public interface IDataHandler
```
Удаляем интерфейс.

MAXon28 commented on LifeHabitTracker/LifeHabitTracker.BusinessLogicLayer/Interfaces/Chain/ITypeHandler.cs:
```diff
@@ -0,0 +1,25 @@
+    /// <summary>
+    /// Функционал обработчика событий
+    /// </summary>
+    public interface ITypeHandler
```
Удаляем интерфейс.

SAWINBEAST committed:
Исправление комментариев

Commit:
LHT-2 Убрал лишние комментарии

Для методов, которые возвращают просто Task нет смысла комментировать, что возвращается (потому что по сути кроме самого таска ничего не возвращается)
